src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -164,9 +165,17 @@ public DynamicACLFilter(final UserProjectCache cache, final Settings settings, f
         this.esClient = TransportClient.builder().settings(clientSettings).addPlugin(SearchGuardSSLPlugin.class)
                 .addPlugin(SearchGuardPlugin.class) // needed for config update
                                                     // action only
-                .build().addTransportAddress(new InetSocketTransportAddress(new InetSocketAddress("localhost", 9300)));
+                .build();
+        try {
+            this.esClient.addTransportAddress(new InetSocketTransportAddress(new InetSocketAddress("localhost", 9300)));
+        }catch(Throwable e) {
+            if(StringUtils.contains(StringUtils.defaultIfEmpty(e.getMessage(), ""), "Connection refused")){
+                this.logger.warn("Cluster may still be initializing. Please be patient: {}", e.getMessage());
+            }
+            throw e;
```

Comment:
Should have been an early return, but now catching specific exception and removed.